crates/node/p2p/src/gossip/builder.rs

```diff
@@ -8,54 +8,56 @@ use libp2p::{
     noise::Config as NoiseConfig, tcp::Config as TcpConfig, yamux::Config as YamuxConfig,
 };
 use std::time::Duration;
-use tokio::sync::watch::Receiver;
+use tokio::sync::watch::{self};
 
 use crate::{
     Behaviour, BlockHandler, GossipDriver, GossipDriverBuilderError, gossip::gater::GaterConfig,
 };
 
 /// A builder for the [`GossipDriver`].
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct GossipDriverBuilder {
-    /// The idle connection timeout as a [`Duration`].
-    timeout: Option<Duration>,
+    /// The [`RollupConfig`] for the network.
+    rollup_config: RollupConfig,
```

Comment:
Having an owned `RollupConfig` feels like tech debt.
It looks like the `RollupConfig` is passed into the `BlockHandler` constructor so it can switch block topics based on hardfork timestamps.
Let me think more about how we can best decouple this logic.
Perhaps we'll want to move the `BlockHandler` into a new `kona-topics` crate that holds the block topic + handler logic so the large `RollupConfig` doesn't need to be passed around like this in memory.